### PR TITLE
Improve JSX documentation

### PR DESCRIPTION
**Pull request by GlassBricks**
Notes:

 - TSX is just the name of the file extension, the syntax is still called JSX (like in here: https://www.typescriptlang.org/docs/handbook/jsx.html). Links may need to be updated for this (in changelog/release notes).
 - Added section on creating your own JSX, and the caveats of that.

### CODE REVIEW

File: docs/jsx.md [hunk 1]
```diff
@@ -0,0 +1,138 @@
...
+
+```lua
+local element = MyNamespace.myCreate(MyNamespace.MyFragment);
+```
+
+## Creating your own JSX
+
+### JSX typings
+
+The types on the jsx factory function itself does _not_ affect how typescript checks JSX types, and no type checking against the jsx factory function is done during transformation.
```
Comment (Perryvw) [inline docs/jsx.md:111]:
```suggestion
The types on the jsx factory function itself do _not_ affect how typescript checks JSX types, and no type checking against the jsx factory function is done during transformation.
```

File: docs/jsx.md [hunk 2]
```diff
@@ -0,0 +1,138 @@
...
+function createElement(
+  type: string | Function | Class,
+  props?: object,
+  ...children: any[]
+): any
+```
+
+- The function should have a `@noSelf` annotation or have a `this: void` parameter. See [here](the-self-parameter.md) for more info.
+- `type` will be a string for intrinsic properties (tag name starts with a lowercase letter), or a function/class component.
+- `props` will be the tag properties as an object/table, or `null`/`nil` if no properties are specified.
```
Comment (Perryvw) [inline docs/jsx.md:130]:
```suggestion
- `props` will be the tag properties as an object/table, or `undefined`/`null`/`nil` if no properties are specified.
```

File: docs/jsx.md [hunk 3]
```diff
@@ -0,0 +1,138 @@
...
+- The remaining parameters form the `children`, and should be collected with a rest parameter (`...`), and not as one array parameter. The type of the children will be strings for inner text, and values passed directly for JSX expressions and nested elements.
+    - No transformations are done on the children parameters, meaning they may have any type (including an arrays) that you may need to handle.
```
Comment (Perryvw) [inline docs/jsx.md:132]:
```suggestion
    - No transformations are done on the children parameters, meaning they may have any type (including arrays) that you may need to handle.
```

**Comment by GlassBricks** [comment]
Don't mind me misclicking the "close pull request" button...

**Pull request merged**